mlir/test/lit.cfg.py

```diff
@@ -18,7 +18,14 @@
 # name: The name of this test suite.
 config.name = "MLIR"
 
-config.test_format = lit.formats.ShTest(not llvm_config.use_lit_shell)
+# We prefer the lit internal shell which provide a better user experience on failures
+# Unless the user explicitly disable it with LIT_USE_INTERNAL_SHELL=0 env var.
+use_lit_shell = True
+lit_shell_env = os.environ.get("LIT_USE_INTERNAL_SHELL")
+if lit_shell_env:
+  use_lit_shell = not lit.util.pythonize_bool(lit_shell_env)
+
+config.test_format = lit.formats.ShTest(execute_external=not use_lit_shell)
```

Comment:
This won't honor the CMake config flag, but it'll respect the environment flag allowing to disable it if needed. Seems like the best I can do right now?

Reply:
What cmake config flag do you mean?

Reply:
Actually I misremembered, it is controlled by the env var always...

Something to note is that we will still have `features.add("shell")` added by lit already here.

Reply:
While I haven't tried, I believe you can just add/remove it in `config.available_features` in this `lit.cfg.py`... in case someone actually tries to use it one day in MLIR.  That's probably for MLIR people to decide.